x-pack/plugins/apm/public/components/shared/select_with_combo_box/index.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiComboBoxOptionOption } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ENVIRONMENT_ALL } from '../../../../common/environment_filter_values';
+
+export const ALL_OPTION_VALUE = 'ALL_OPTION_VALUE';
+
+export const allOptionText = i18n.translate('xpack.apm.allOption', {
+  defaultMessage: 'All',
+});
+
+export const allOption: EuiComboBoxOptionOption<string> = {
+  label: allOptionText,
+  value: ALL_OPTION_VALUE,
+};
+
+export const environmentAllOption: EuiComboBoxOptionOption<string> = {
+  label: ENVIRONMENT_ALL.text,
+  value: ENVIRONMENT_ALL.value,
+};
```

Comment:
it would be nice if `ENVIRONMENT_ALL.text` was renamed to `ENVIRONMENT_ALL.label` in `apm/common/environment_filter_values.ts`. then it would already match the option type.

Reply:
Good point. That way I can reduce the duplicates and reuse ENVIRONMENT_ALL